app/src/main/java/com/android/sample/ui/listActivities/LikedActivities.kt

```diff
@@ -0,0 +1,254 @@
+  }
+
+  Card(
+      modifier =
+          Modifier.fillMaxWidth()
+              .testTag("activityCard")
+              .clip(RoundedCornerShape(16.dp))
+              .clickable {
+                listActivitiesViewModel.selectActivity(activity)
+                navigationActions.navigateTo(Screen.ACTIVITY_DETAILS)
+              },
+      elevation = CardDefaults.cardElevation(8.dp)) {
+        Column {
+          // Box for overlaying the title on the image
+          Box(modifier = Modifier.fillMaxWidth().height(180.dp)) {
+            // Display the activity image
+            Image(
+                painter = painterResource(R.drawable.foot),
+                contentDescription = activity.title,
+                modifier = Modifier.fillMaxWidth().height(180.dp),
+                contentScale = ContentScale.Crop)
+
+            // Display the activity name on top of the image
+            Text(
+                text = activity.title,
+                style =
+                    MaterialTheme.typography.titleLarge.copy(
+                        fontWeight = FontWeight.Bold,
+                        color = Color.White // Title color set to black
+                        ),
+                modifier = Modifier.align(Alignment.BottomStart).padding(16.dp))
+          }
+
+          Spacer(modifier = Modifier.height(8.dp))
+          Row(
+              modifier = Modifier.padding(horizontal = 16.dp).fillMaxWidth(),
+              horizontalArrangement = Arrangement.SpaceBetween,
+              verticalAlignment = Alignment.CenterVertically) {
+                // Display the date
+                Text(
+                    text = formattedDate,
+                    style =
+                        MaterialTheme.typography.bodySmall.copy(
+                            color = Color.Gray, // Light gray color for the date
+                            fontStyle = FontStyle.Italic),
+                    modifier = Modifier.weight(1f) // Takes up remaining space
+                    )
+
+                if (profile != null) {
+                  IconButton(
+                      modifier = Modifier.testTag("favoriteIcon$isLiked"),
+                      onClick = {
+                        isLiked = !isLiked
+                        if (isLiked) {
+                          profileViewModel.addLikedActivity(profile.id, activity.uid)
+                        } else {
+                          profileViewModel.removeLikedActivity(profile.id, activity.uid)
+                        }
+                      },
+                  ) {
+                    Icon(
+                        imageVector =
+                            if (isLiked) Icons.Filled.Favorite else Icons.Outlined.FavoriteBorder,
+                        contentDescription = if (isLiked) "Liked" else "Not Liked",
+                        tint = if (isLiked) Color.Black else Color.Gray)
+                  }
+                }
+              }
```

Comment:
The IconButton changes dynamically based on the isLiked state, which provides clear visual feedback to the user about the current status of an activity in their liked list. This immediate feedback is essential for a good user experience.